infra/base-images/base-builder/compile_javascript_fuzzer

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash -eu
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+project=$1
+# Path the fuzz target source file relative to the project's root
+fuzz_target=$2
+# Arguments to pass to Jazzer.js
+jazzerjs_args=${@:3}
+
+# Copy source code into the $OUT directory.
+if [ ! -d $OUT/$project ]; then
+  cp -r $SRC/$project $OUT/$project
+fi
+
+fuzzer_basename=$(basename -s .js $fuzz_target)
+fuzzer_dir=$(dirname $fuzz_target)
+
+# Create an execution wrapper that executes Jazzer.js with the correct arguments.
+echo "#!/bin/bash
+# LLVMFuzzerTestOneInput so that the wrapper script is recognized as a fuzz target for 'check_build'.
+cd $project/$fuzzer_dir
+jazzer $fuzzer_basename $jazzerjs_args -- \$@" > $OUT/$fuzzer_basename
```

Comment:
Does this rely on `jazzer` existing in `PATH` ? Is there a way to bundle this together with the build?

Reply:
Yes. However, this is guaranteed once we install Jazzer.js globally using `npm`. We can bundle Jazzer.js in the build by installing it locally into the project after copying the source code into `$OUT`. We can then start Jazzer.js CLI using `npx jazzer`. I've pushed a commit that does that.